orchagent/switchorch.cpp

```diff
@@ -77,12 +120,15 @@ SwitchOrch::SwitchOrch(DBConnector *db, vector<TableConnector>& connectors, Tabl
         m_db(db),
         m_stateDb(new DBConnector(STATE_DB, DBConnector::DEFAULT_UNIXSOCKET, 0)),
         m_asicSensorsTable(new Table(m_stateDb.get(), ASIC_TEMPERATURE_INFO_TABLE_NAME)),
-        m_sensorsPollerTimer (new SelectableTimer((timespec { .tv_sec = DEFAULT_ASIC_SENSORS_POLLER_INTERVAL, .tv_nsec = 0 })))
+        m_sensorsPollerTimer (new SelectableTimer((timespec { .tv_sec = DEFAULT_ASIC_SENSORS_POLLER_INTERVAL, .tv_nsec = 0 }))),
+        m_stateDbForNotification(new DBConnector(STATE_DB, DBConnector::DEFAULT_UNIXSOCKET, 0)),
+        m_asicSdkHealthEventTable(new Table(m_stateDbForNotification.get(), STATE_ASIC_SDK_HEALTH_EVENT_TABLE_NAME))
 {
     m_restartCheckNotificationConsumer = new NotificationConsumer(db, "RESTARTCHECK");
     auto restartCheckNotifier = new Notifier(m_restartCheckNotificationConsumer, this, "RESTARTCHECK");
     Orch::addExecutor(restartCheckNotifier);
 
+    initAsicSdkHealthEventNotification();
```

Comment:
Do we have a global config knob for this feautre? IMO, this should not be called from constructor and start lua if the feature is disabled but ASIC supports it

Reply:
Yes, we do, but it is enabled by default.
Do you think we need to check the DB and initialize it only if it is enabled?
I am OK to do so, but it seems to be a common logic to initialize a feature without checking the configuration.

Reply:
I see there are such examples to test whether the device is VoQ etc. So, I'll fix it in this way.

Reply:
Fixed.